### PR TITLE
EC2 provider - collect VPCs data

### PR DESCRIPTION
**Pull request by uri-weisman**
### Summary of your changes
1. Extend EC2 provider to collect VPC data.
2. Minor changes to the s3 provider - ACL resource format.
3. Network factory - fix a bug in creating the multi-clients.

### Related Issues
- Related: https://github.com/elastic/csp-security-policies/pull/180

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: resources/fetchers/network_fetcher.go [hunk 1]
```diff
@@ -48,22 +47,11 @@ type NetworkResource struct {
...
-	nacl, err := awslib.MultiRegionFetch(ctx, f.ec2Clients, func(ctx context.Context, client ec2.ElasticCompute) ([]awslib.AwsResource, error) {
-		return client.DescribeNetworkAcl(ctx)
-	})
-	if err != nil {
-		f.log.Errorf("failed to describe network acl: %v", err)
-	}
-
-	securityGroups, err := awslib.MultiRegionFetch(ctx, f.ec2Clients, func(ctx context.Context, client ec2.ElasticCompute) ([]awslib.AwsResource, error) {
-		return client.DescribeSecurityGroups(ctx)
+	resources, _ := awslib.MultiRegionFetch(ctx, f.ec2Clients, func(ctx context.Context, client ec2.ElasticCompute) ([]awslib.AwsResource, error) {
```
Comment (olegsu) [inline resources/fetchers/network_fetcher.go:50]: Awesome 🔥

File: resources/fetchers/network_fetcher.go [hunk 2]
```diff
@@ -91,4 +79,36 @@ func (r NetworkResource) GetMetadata() (fetching.ResourceMetadata, error) {
...
+	vpcs, err := client.DescribeVPCs(ctx)
+	if err != nil {
+		f.log.Errorf("failed to describe vpcs: %v", err)
+	}
+	resources = appendList(resources, vpcs)
+
+	return resources, nil
+}
+
+func appendList(list []awslib.AwsResource, resources []awslib.AwsResource) []awslib.AwsResource {
```
Comment (olegsu) [inline resources/fetchers/network_fetcher.go:108]: Why this is needed?
append to nil is not harmful
```go
	r1 := []*Data{{}, {}} // slice with two items
	var r2 []*Data // zero value of slice is nil
	r3 := append(r1, r2...)
	fmt.Println(r2 == nil)
	fmt.Print(len(r3), cap(r3))
```

Will output
```bash
true
2 2
```

File: resources/providers/awslib/multi_region.go [hunk 3]
```diff
@@ -71,7 +71,7 @@ func (w *MultiRegionClientFactory[T]) NewMultiRegionClients(client DescribeCloud
 	return wrapper
 }
 
-// Fetch retrieves resources from multiple regions concurrently using the provided fetcher function.
+// MultiRegionFetch retrieves resources from multiple regions concurrently using the provided fetcher function.
```
Comment (olegsu) [inline resources/providers/awslib/multi_region.go:74]: Thank you

File: resources/fetchers/network_factory.go [hunk 4]
```diff
@@ -71,7 +71,7 @@ func (f *EC2NetworkFactory) CreateFrom(log *logp.Logger, cfg ACLFetcherConfig, c
 		return nil, fmt.Errorf("could not get cloud indentity: %w", err)
 	}
 	ff := func(cfg awssdk.Config) ec2.ElasticCompute {
```
Comment (olegsu) [inline resources/fetchers/network_factory.go:73]: 👍

**Review by olegsu — CHANGES_REQUESTED** [review]

File: resources/fetchers/network_fetcher.go (hunk 2 above)
Comment (uri-weisman) [reply resources/fetchers/network_fetcher.go]: reverted

**Review by olegsu — APPROVED** [review]

**Pull request merged**